src/ui/public/courier/fetch/call_client.js

```diff
@@ -106,28 +122,27 @@ export function CallClientProvider(Private, Promise, es) {
     // We're going to create a new async context here, so that the logic within it can execute
     // asynchronously after we've returned a reference to defer.promise.
     Promise.resolve().then(async () => {
-      // Flatten the searchSource within each searchRequest to get the fetch params,
-      // e.g. body, filters, index pattern, query.
-      const allFetchParams = await getAllFetchParams(requestsToFetch);
-
-      // Serialize the fetch params into a format suitable for the body of an ES query.
-      const serializedFetchParams = await serializeAllFetchParams(allFetchParams, requestsToFetch);
-
-      // The index of the request inside requestsToFetch determines which response is mapped to it.
-      // If a request won't generate a response, since it already failed, we need to remove the
-      // request from the requestsToFetch array so the indexes will continue to match up to the
-      // responses correctly.
-      requestsToFetch = requestsToFetch.filter(request => request !== undefined);
+      // Execute each request using its search strategy.
+      const esPromises = searchStrategiesWithRequests.map(searchStrategyWithSearchRequests => {
+        const { searchStrategy, searchRequests } = searchStrategyWithSearchRequests;
+        return searchStrategy.search({ searchRequests, es, Promise, serializeFetchParams });
+      });
 
       try {
         // The request was aborted while we were doing the above logic.
         if (isRequestAborted) {
           throw ABORTED;
         }
 
-        esPromise = es.msearch({ body: serializedFetchParams });
-        const clientResponse = await esPromise;
-        await respond(clientResponse.responses);
+        esPromise = Promise.all(esPromises);
```

Comment:
Also, would you mind trying to write a test that validates that promises from the esClient are aborted correctly when a single request in a batch is aborted? I imagine it's going to be tricky but would probably be a good idea to have given the scope of this refactor.

Reply:
> We'll need to update the esPromise logic to handle the esPromises array because it needs access to the individual esPromise.abort() methods

Can you see any problem with just removing the code that calls `esPromise.abort()`? What's the point of aborting them?